src/bsoncxx/json.cpp

```diff
@@ -42,247 +42,20 @@ void bson_free_deleter(std::uint8_t* ptr) {
-
-        pad();
-        out << "}";
-    }
-
-    void visit_value(const types::b_code& value) {
-        out << value.code.to_string().data();
-    }
-
-    void visit_value(const types::b_symbol& value) {
-        out << value.symbol.to_string().data();
-    }
-
-    void visit_value(const types::b_codewscope& value) {
-        out << value.code.to_string().data();
-    }
-
-    void visit_value(const types::b_int32& value) {
-        out << value.value;
-    }
-
-    void visit_value(const types::b_timestamp& value) {
-        out << "{\n";
-        pad(1);
-        out << "\"$timestamp\" : {\n";
-        pad(2);
-        out << "\"$t\" : " << value.timestamp << ",\n";
-        pad(2);
-        out << "\"$i\" : " << value.increment << '\n';
-        pad(1);
-        out << "}";
-        pad();
-        out << "}";
-    }
-
-    void visit_value(const types::b_int64& value) {
-        out << value.value;
-    }
-
-    void visit_value(const types::b_minkey&) {
-        out << "{\n";
-        pad(1);
-        out << "\"$minKey\" : 1\n";
-        pad();
-        out << "}";
-    }
-
-    void visit_value(const types::b_maxkey&) {
-        out << "{\n";
-        pad(1);
-        out << "\"$maxKey\" : 1\n";
-        pad();
-        out << "}";
-    }
-
-   private:
-    std::ostream& out;
-    std::vector<bool> stack;
-    std::size_t padding;
-
-    void pad(std::size_t extra = 0) {
-        out << std::setw((stack.size() + -1 + extra + padding) * 4) << "" << std::setw(0);
-    }
-
-    void visit_children(const document::view& view) {
-        bool first = true;
-        for (auto&& x : view) {
-            if (!x) {
-                break;
-            }
-            if (!first) {
-                out << ", \n";
-            }
-            first = false;
-            visit_key(x.key());
-            switch (static_cast<int>(x.type())) {
-#define BSONCXX_ENUM(name, val)      \
-    case val:                        \
-        visit_value(x.get_##name()); \
-        break;
-#include <bsoncxx/enums/type.hpp>
-#undef BSONCXX_ENUM
-            }
-        }
-        out << '\n';
-        stack.pop_back();
-    }
-};
-
 }  // namespace
 
 std::string to_json(document::view view) {
-    std::stringstream ss;
-
-    json_visitor v(ss, false, 0);
-    v.visit_value(types::b_document{view});
-
-    return ss.str();
-}
-
-std::string to_json(document::element element) {
-    std::stringstream ss;
-
-    json_visitor v(ss, false, 0);
-
-    switch ((int)element.type()) {
-#define BSONCXX_ENUM(name, val)              \
-    case val:                                \
-        v.visit_key(element.key());          \
-        v.visit_value(element.get_##name()); \
-        break;
-#include <bsoncxx/enums/type.hpp>
-#undef BSONCXX_ENUM
-    }
-
-    return ss.str();
-}
-
-std::string to_json(types::value value) {
-    std::stringstream ss;
+    bson_t bson;
+    bson_init_static(&bson, view.data(), view.length());
```

Comment:
check the return value on init_static and throw if the view isn't valid?  That'll catch invalid and mismatched lengths.

Otherwise an invalid view will lead you to bson_as_json some uninitialized bytes